### DISCUSSION TITLE
Shorten the current directory title.

### DISCUSSION THREAD

**Discussion by madelaney** [nvim-neo-tree/neo-tree.nvim#681] [General]
When I open neovim (0.8.2), and neo-tree (v2.x@3b41f0d17139).

When I open a directory, I was wondering if it is possible to shorten the path. For example, rather then displaying `~/work/gitlab.mldelaney.io/project/sub-group/repo`, it would display `~/w/g/p/s/repo` or something similar.

I looked over the docs, and I couldn't find anything but I'm new to both neovim and neo-tree so perhaps I just missed it.

**Comment by nyngwang** [comment]
FYI: you could press `e` to toggle-expand the width to see the full path.

**Reply by madelaney** [reply]
Thanks, but that's not really what I was looking for. I want to shorten the path, not display the fully qualified path.

**Comment by pysan3** [comment]
Unfortunately, it does not seem to be possible with the current situation. (I also would recommend using the `e` keybind ~~cuz I implemented it XD~~ tho).

At least, the root name comes from [this line](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8238865e1d9c61f1a260c290653f2c419503e0a9/lua/neo-tree/sources/filesystem/lib/fs_scan.lua#L386).

So if you @madelaney are interested, it would be nice if you could implement a function to calculate the shortened path and stick it in here.

**Comment by caenrique** [comment]
I would also really appreciate this feature, even something as simple as allowing to specify the expression for `vim.fn.fnamemodify()` in the config.
I looked into the linked line and changing `:~` for `:t` does the trick for me, but I couldn't figure out what the proper way of getting that from config would be

**Comment by cseickel** [comment]
The original intention behind neo-tree is that if you want to customize it, there are hooks to do whatever you want, but we are not going to add endless configuration options.

The hook in this case is to replace the `name` component with your own function that displays the root however you'd like. The only problem with that theory is that the documentation is not great. Here's an example for what you want to do:

```
      local fc = require("neo-tree.sources.filesystem.components")

      require('neo-tree').setup({
        filesystem = {
          components ={
            name = function(config, node, state)
              local result = fc.name(config, node, state)
              if node:get_depth() == 1 and node.type ~= "message" then
                result.text = vim.fn.fnamemodify(node.path, ":t")
              end
              return result
            end,
          },
        }
      })
```

This just calls the built-in name component and then overrides the name if it is the root node.

See `:h neo-tree-components` for more information.

**Reply by caenrique** [reply]
That's perfect. Thanks!
